From 641d647e21bc66e098b231c1c2654890540356e3 Mon Sep 17 00:00:00 2001
From: Francesco Bagagli <f.bagagli@evidence.eu.com>
Date: Fri, 4 Oct 2019 15:34:47 +0200
Subject: [PATCH 1/1] fix packaging:

* add requirements_dev.txt: must be installed in the virtualenv
                            used for development
* add requirements_test.txt: requirements used for testing
                             pytest, coverage...
* add makefile:
* add support for tox: tox.ini and pyproject.toml
---
 .gitignore            |  6 ++++++
 MANIFEST.in           |  7 +++++++
 Makefile              | 33 +++++++++++++++++++++++++++++++++
 pyproject.toml        |  3 +++
 requirements_dev.txt  |  6 ++++++
 requirements_test.txt |  1 +
 setup.py              | 13 ++++++++++---
 tox.ini               |  8 ++++++++
 8 files changed, 74 insertions(+), 3 deletions(-)
 create mode 100644 .gitignore
 create mode 100644 MANIFEST.in
 create mode 100644 Makefile
 create mode 100644 pyproject.toml
 create mode 100644 requirements_dev.txt
 create mode 100644 requirements_test.txt
 create mode 100644 tox.ini

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..d866243
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,6 @@
+venv
+.tox
+test-result
+*.pyc
+__pycache__
+*.egg-info
diff --git a/MANIFEST.in b/MANIFEST.in
new file mode 100644
index 0000000..93fee72
--- /dev/null
+++ b/MANIFEST.in
@@ -0,0 +1,7 @@
+include LICENSE.txt
+include README.md
+include tox.ini
+
+recursive-include tests *
+include requirements.txt
+include requirements_dev.txt
diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..75eb6c4
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,33 @@
+.PHONY: clean clean-test clean-pyc clean-build docs help
+
+clean: clean-build clean-pyc clean-test ## remove all build, test, and Python artifacts
+
+clean-build: ## remove build artifacts
+	rm -fr build/
+	rm -fr dist/
+	rm -fr .eggs/
+	find . -name '*.egg-info' -exec rm -fr {} +
+	find . -name '*.egg' -exec rm -f {} +
+
+clean-pyc: ## remove Python file artifacts
+	find . -name '*.pyc' -exec rm -f {} +
+	find . -name '*.pyo' -exec rm -f {} +
+	find . -name '*~' -exec rm -f {} +
+	find . -name '__pycache__' -exec rm -fr {} +
+
+clean-test: ## remove test and coverage artifacts
+	rm -fr .tox/
+	rm -f .coverage
+	rm -fr htmlcov/
+	rm -fr .pytest_cache
+
+test:
+	tox
+
+dist: clean ## builds source and wheel package
+	python setup.py sdist
+	python setup.py bdist_wheel
+	ls -l dist
+
+upload: dist
+	twine upload dist/*
diff --git a/pyproject.toml b/pyproject.toml
new file mode 100644
index 0000000..b8d6ec0
--- /dev/null
+++ b/pyproject.toml
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["wheel >= 0.33", "setuptools >= 40.0.0", "pip >= 19.2.3"]
+build-backend = "setuptools.build_meta"
diff --git a/requirements_dev.txt b/requirements_dev.txt
new file mode 100644
index 0000000..2fa0f71
--- /dev/null
+++ b/requirements_dev.txt
@@ -0,0 +1,6 @@
+tox>=3.14
+pip>=19.2.3
+wheel>=0.33
+setuptools>=40.0.0
+pip>=19.2.3
+twine>=2.0.0
diff --git a/requirements_test.txt b/requirements_test.txt
new file mode 100644
index 0000000..2a92095
--- /dev/null
+++ b/requirements_test.txt
@@ -0,0 +1 @@
+pytest~=5.2
diff --git a/setup.py b/setup.py
index 53bd3ed..308f813 100644
--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-import setuptools
+from setuptools import setup, find_packages
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
@@ -6,7 +6,7 @@ with open("README.md", "r") as fh:
 with open("requirements.txt") as f:
     required_list = f.read().splitlines()
 
-setuptools.setup(
+setup(
     name="filterpicker",
     version="1.0.1",
     author="Matteo Bagagli",
@@ -17,11 +17,18 @@ setuptools.setup(
     url="https://gitlab.ethz.ch/mbagagli/filterpicker",
     python_requires='>=3.6',
     install_requires=required_list,
-    packages=setuptools.find_packages(),
+    setup_requires=['wheel'],
+    packages=find_packages(),
     classifiers=[
         "Programming Language :: Python :: 3",
         "License :: OSI Approved :: MIT License",
         "Operating System :: Unix",
         "Intended Audience :: Science/Research",
     ],
+    # entry_points={
+    #     'console_script': [
+    #         'command_name='
+    #         'module, that implement this',
+    #     ],
+    # },
 )
diff --git a/tox.ini b/tox.ini
new file mode 100644
index 0000000..69bdd0c
--- /dev/null
+++ b/tox.ini
@@ -0,0 +1,8 @@
+[tox]
+envlist = py36, py37
+isolated_build = True
+
+[testenv]
+deps =
+    -rrequirements_dev.txt
+commands = pytest tests
-- 
2.17.1

